lib/gitlab/ci/templates/jobs/docker/Kaniko.yml

```diff
@@ -51,3 +53,48 @@ kaniko_publish:
     - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
     - if: $CI_PIPELINE_SOURCE == "web"
     - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+
+# kaniko_publish_arm: Extends the standard Kaniko publish job for ARM-based architectures.
+# This job inherits all logic from `.kaniko_publish` and customizes execution by using a specific GitLab runner tag.
+# The tag `architecture-aarch64` ensures that the job runs on an ARM64 (AArch64) compatible GitLab runner.
+#
+# Use Case:
+# - When building Docker images for ARM-based environments such as AWS Graviton, Apple M1/M2 chips, or Jetson devices.
+# - Ensures proper architecture targeting during multi-arch image builds or ARM-specific deployments.
+kaniko_publish_arm:
+  stage: publish
+  extends: .kaniko_publish
+  tags:
+    - architecture-aarch64
+  rules:
+    - when: always
+
+# Standalone multiarch manifest job with its own variables
+multiarch_manifest_publish:
```

Comment:
Please add the new kaniko_publish_arm and multiarch_manifest_publish Gitlab jobs to the README.md. See https://github.com/chesapeaketechnology/gitlab-templates/blob/release/4/README.md#kaniko-docker-image-publishing-job for an example of how the documentation for a Gitlab job is typically written for this Github repo.